### PR TITLE
Allow restore db snapshot from snapshot 

### PR DESCRIPTION
**Pull request by brucegucode**
Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/1390

Description of changes:
It's similar to what we did in https://github.com/aws-controllers-k8s/rds-controller/pull/78 
This PR will enable restore db cluster from snapshot, either dbSnapshot or dbClusterSnapshot. 

Fixed the issue during testing and verified it can work with both cluster snapshot and db instance snapshot. Controller can issue restore db cluster from snapshot API and resource is synced after creation succeed.  

For cluster snapshot restore to db cluster: 
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBCluster

metadata:
  name: "brucegu-restore-0"

spec:
  backupRetentionPeriod: 1
  dbClusterIdentifier: brucegu-restore-0
  engine: aurora-postgresql
  engineVersion: "13.7"
  snapshotIdentifier: brucegu-cluster-snapshot-0
```

For db instance snapshot restore to db cluster: (i have to use ARN here according to https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_RestoreDBClusterFromSnapshot.html )
```
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBCluster

metadata:
  name: "brucegu-restore-2"

spec:
  backupRetentionPeriod: 1
  dbClusterIdentifier: brucegu-restore-2
  engine: aurora-postgresql
  engineVersion: "13.7"
  snapshotIdentifier: arn:aws:rds:us-west-2:316992738638:snapshot:brucegu-restore-instance-0
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by brucegucode** [comment]
test failing due to below issue , looks like to be test-infra related ? 
```
./scripts/run-helm-tests.sh: line 27: /home/prow/go/src/github.com/aws-controllers-k8s/test-infra/scripts/lib/../start.sh: No such file or directory
make: *** [Makefile:20: kind-helm-test] Error 1
```

File: helm/values.schema.json [hunk 1]
```diff
@@ -192,7 +192,7 @@
       "type": "array",
       "items": {
         "type": "string",
-        "pattern": "^.*=.*$"
+        "pattern": "(^$|^.*=.*$)"
```
Comment (jaypipes) [inline helm/values.schema.json:195]: hmm, this seems like an odd change, @brucegucode. Is this related to DB cluster restore from snapshot?

File: generator.yaml [hunk 2]
```diff
@@ -120,6 +124,10 @@ resources:
           resource: SecurityGroup
           service_name: ec2
           path: Status.ID
+      SnapshotIdentifier:
+        from:
+          operation: RestoreDBClusterFromSnapshot
+          path: SnapshotIdentifier
```
Comment (jaypipes) [inline generator.yaml:130]: What happens if the user changes the value of `Spec.SnapshotIdentifier`? :)

**Review by jaypipes — COMMENTED** [review]
@brucegucode this looks good to me. I do think, however, a short tutorial in our docs on how to create DB instances and DB clusters from a snapshot would be useful to RDS ACK users In addition, potentially a callout in that doc/tutorial that indicates what the expected behaviour is if the user ends up updating the `Spec.SnapshotIdentifier` for a DBCluster or DBInstance. I believe the expected behaviour is that nothing will happen at all, but that might not meet the *user's* expectations :)

File: helm/values.schema.json (hunk 1 above)
Comment (brucegucode) [reply helm/values.schema.json:195]: this was not related to this pr, it comes from generating code with latest code-generator. https://github.com/aws-controllers-k8s/code-generator/commit/cb4bca850e1d1dd678d8ae9bd9d5fd537c06ca51

File: generator.yaml (hunk 2 above)
Comment (brucegucode) [reply generator.yaml:130]: Yeah, like you said, it will end up with nothing happening if customer update it. 
I will add a tutorial part about how customer do restore db instance/cluster from snapshot and mention customer need be careful not to update this `SnapshotIdentifier` after it's set

File: helm/values.schema.json (hunk 1 above)
Comment (jaypipes) [reply helm/values.schema.json:195]: Got it. :)

**Comment by brucegucode** [comment]
e2e test failed due to some bootstrap logic
I'm going to override it since this code won't impact testing logic. 
```
2022-07-19T16:53:22+00:00 [INFO] Running test bootstrap ...
Traceback (most recent call last):
  File "service_bootstrap.py", line 20, in <module>
    from acktest import resources
  File "/usr/local/lib/python3.8/site-packages/acktest/resources.py", line 48, in <module>
    in_str: str, replacement_dictionary: Dict[str, Any] = default_placeholder_values()) -> str:
  File "/usr/local/lib/python3.8/site-packages/acktest/resources.py", line 32, in default_placeholder_values
    "AWS_ACCOUNT_ID": identity.get_account_id(),
  File "/usr/local/lib/python3.8/site-packages/acktest/aws/identity.py", line 20, in get_account_id
    return boto3.client('sts').get_caller_identity().get('Account')
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 662, in _make_api_call
    http, parsed_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 682, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
```
